### PR TITLE
Prepare for prow onboarding

### PR DESCRIPTION
**Pull request by jlpadilla**
**Related Issue:**  open-cluster-management/backlog#10047

### Description of changes
- Add Makefile.prow
- Add Dockerfile.prow
- Update Dockerfile to be consistent with Dockerfile.prow
- Upgrade to Go 1.16
- Add DCO

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
-go 1.15
+go 1.16
 
 require (
-	cloud.google.com/go v0.54.0 // indirect
-	github.com/Azure/go-autorest/autorest v0.11.1 // indirect
-	github.com/Azure/go-autorest/autorest/adal v0.9.5 // indirect
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	github.com/golangci/golangci-lint v1.24.0 // indirect
```
Comment (mqhackett) [inline go.mod:10]: removed lint from require? Was that intentional, or the result of a `go mod build` operation

**Review by mqhackett — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (jlpadilla) [reply go.mod:10]: intentional because it's only used by the build lint step and not the product.

File: Dockerfile [hunk 2]
```diff
@@ -35,14 +44,11 @@ RUN microdnf install ca-certificates vi --nodocs &&\
...
 
-
-
-ADD output/insights-client /bin
-
-
 EXPOSE 3030
 USER ${USER_UID}
-ENTRYPOINT ["/bin/insights-client"]
+ENTRYPOINT ["/bin/main"]
```
Comment (mqhackett) [inline Dockerfile:54]: What is the rationale behind switching our Dockerfile entrypoints from _project name_ to _main_ ? I appreciate the uniformity, just curious.

Comment (jlpadilla) [reply Dockerfile:54]: just for consistency

File: Dockerfile.prow [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      summary="$IMAGE_SUMMARY" \
+      io.k8s.display-name="$IMAGE_DISPLAY_NAME" \
+      io.k8s.description="$IMAGE_DESCRIPTION" \
+      io.openshift.tags="$IMAGE_OPENSHIFT_TAGS"
+
+RUN microdnf install ca-certificates vi --nodocs &&\
+    mkdir /licenses &&\
+    microdnf clean all
+
+COPY --from=builder /go/src/github.com/open-cluster-management/insights-client/main /bin/main
```
Comment (xrajesh) [inline Dockerfile.prow:47]: @jlpadilla Is this path correct ? We build the binary in output folder

Comment (jlpadilla) [reply Dockerfile.prow:47]: Yes, the binary is being built inside the "builder" docker image. (line 8 above)

That's different from `make build` which builds the binary in the local  system, or travis worker node.

Comment (xrajesh) [reply Dockerfile.prow:47]: 👍

**Review by xrajesh — APPROVED** [review]

**Pull request merged**